src/main/java/com/patchworkmc/mapping/remapper/NaiveRemapper.java

```diff
@@ -75,11 +85,22 @@ public String getClass(String volde) {
 		return classes.getOrDefault(volde, volde);
 	}
 
+	/**
+	 * @deprecated Because of java generics, inheritence, synthetics, and recompiling, Forge matches methods that override another method but change
+	 *  return type T to class_XXX directly, but Fabric matches to the synthetic. Runtime mappings should be used here instead.
+	 * @param volde
+	 * @return
+	 */
+	@Deprecated
 	public String getMethod(String volde) {
 		if (!volde.startsWith("func_")) {
 			throw new IllegalArgumentException("Cannot remap methods not starting with func_: " + volde);
 		}
 
+		if (blacklistedMethods.contains(volde)) {
+			throw new AssertionError("Cannot remap method " + volde + " due to an issue described in the Javadoc of this method.");
```

Comment:
```suggestion
			throw new UnsupportedOperationException("Cannot remap method name " + volde + " because that method name could map to multiple targets!");
```